### PR TITLE
Separate version properties for hibernate-enhance and quarkus-maven

### LINKED ISSUE #20261: Quarkus 3 build properties break product build
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

dist/quarkus

### Describe the bug

The quarkus3 branch was merged recently. This involved some significant changes to the pom files, which didn't end up in an ideal state for a product build. Particularly with the version properties, certain dependencies/plugins are tied together that shouldn't be.

### Version

21

### Expected behavior

The maven build should complete without errors

### Actual behavior

The build fails after product alignment, due to missing dependencies

### How to Reproduce?

Try changing `hibernate-orm.version`

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ASzc**
This will allow them to be defined seperately from aligned dependency versions suitable for product

Closes #20261

### CODE REVIEW

File: operator/pom.xml
Comment (vmuzikar) [inline operator/pom.xml:39]: The Operator uses an independent Quarkus version from Keycloak. That's because the Operator uses [QOSDK](https://github.com/quarkiverse/quarkus-operator-sdk) which uses Quarkus as a transitive dependency. Hence, we should stick here to whatever Quarkus version does QOSDK [use](https://github.com/quarkiverse/quarkus-operator-sdk/blob/6.0.1/pom.xml#L15). Eventually, once we start using the LTS versions of Quarkus and QOSDK, they should be the same but that's not the case right now.

File: model/map-jpa/pom.xml
Comment (vmuzikar) [inline model/map-jpa/pom.xml:54]: I believe we need a review from @keycloak/store.

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: operator/pom.xml
Comment (ASzc) [reply operator/pom.xml:39]: Hmm. Ok, this should be possible to do, but the operator must use a distinct version property. I will change this PR to implement that

**Review by vmuzikar — DISMISSED** [review]
LGTM from the Cloud Native team perspective, thanks @ASzc!

Let's wait for a review from Store team. :)

File: model/map-jpa/pom.xml
Comment (mhajas) [reply model/map-jpa/pom.xml:54]: I would say it makes sense to have the property defined in the parent pom and then use it here.

**Review by mhajas — CHANGES_REQUESTED** [review]

File: model/map-jpa/pom.xml
Comment (ASzc) [reply model/map-jpa/pom.xml]: Ok, updated in latest push

**Review by vmuzikar — APPROVED** [review]

**Comment by ASzc** [comment]
I think this PR is now the last remaining thing to get the product build green again

**Review by mhajas — APPROVED** [review]
Thanks @ASzc

**Comment by ASzc** [comment]
I had to rebase to allow merging

**Pull request merged**
